libmscore/figuredbass.cpp

```diff
@@ -1770,6 +1773,256 @@ FiguredBass* Score::addFiguredBass()
+      };
+      for (Element* e : segment->annotations()) {
+            if (e->type() == Element::Type::FIGURED_BASS ) {
+                  FiguredBass *fb = static_cast<FiguredBass*>(e);
+                  std::vector<int> harmony;
+                  std::vector<FiguredBassItem*> items = fb->getFiguredBaseItems();
+                  sort( items.begin(), items.end(), [&](FiguredBassItem *i1, FiguredBassItem *i2) {
+                        return get_digit(i1->digit()) < get_digit(i2->digit());
+                  });
+                  FiguredBassItem::Modifier modifier = (none == items[0]->prefix()) ? items[0]->suffix(): items[0]->prefix();
+                  switch ( items.size() ) {
+                        case 0:
+                              render_harmony(none, 3, none);
+                              render_harmony(none, 5, none);
+                              break;
+                        case 1: {
+                              // a single figured bass missing a number, just with a symbol, implies the digit 3.
+                              int digit = (FBIDigitNone==items[0]->digit()) ? 3 : items[0]->digit();
+                              switch (digit) {
+                                    case 2:
+                                          // render_relative({2,4,6});
+                                          render_harmony(modifier,2,none);
+                                          render_harmony(none, 4, none);
+                                          render_harmony(none, 6, none);
+                                    case 3:
+                                          // render_relative({3,5});
+                                          render_harmony(modifier,3,none);
+                                          render_harmony(none, 5, none);
+                                          break;
+                                          //case 4:
+                                    case 5:
+                                          // render_relative({3,5});
+                                          render_harmony(none,3,none);
+                                          render_harmony(modifier, 5, none);
+                                          break;
+                                    case 6:
+                                          // render_relative({3,6});
+                                          render_harmony(none, 3, none);
+                                          render_harmony(modifier, 6, none);
+                                          break;
+                                    case 7:
+                                          // render_relative({3,5,7});
+                                          render_harmony(none,3,none);
+                                          render_harmony(none, 5, none);
+                                          render_harmony(modifier, 7, none);
+                                          break;
+                                    default:
```

Comment:
`8`, `9`, `10` occur and their meaning is rather straightforward (once in a while!): `8` can be ignored (=> `3 5`), `9` = `2`, `10` = `3`. The difference being that they usually require the octave above the bass note.